### PR TITLE
Fusetools2 526 completion for camel component dependency

### PR DESCRIPTION
**Pull request by apupier**
based on https://github.com/camel-tooling/camel-language-server/pull/397

camelComponentDependencyCompletionInModeline

### CODE REVIEW

File: src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		return startPosition <= position && position <= getEndPositionInLine();
+	}
+	
+	@Override
+	public CompletableFuture<List<CompletionItem>> getCompletions(int position, CompletableFuture<CamelCatalog> camelCatalog) {
+		if(getStartPositionInLine() == position) {
+			return camelCatalog.thenApply(catalog -> catalog.findComponentNames().stream()
+				.map(componentName -> ModelHelper.generateComponentModel(catalog.componentJSonSchema(componentName), true))
+				.map(componentModel -> {
+					CompletionItem completionItem = new CompletionItem("camel-"+componentModel.getScheme());
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java:65]: so the dependency takes the scheme? or would it be better to get the dependency instead?

File: src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineOption.java [hunk 2]
```diff
@@ -43,6 +44,8 @@ private ICamelKModelineOptionValue createOptionValue(String option, int nameValu
 			int startPosition = getStartPositionInLine() + optionName.length() + 1;
 			if("trait".equals(optionName)) {
 				return new CamelKModelineTraitOption(value, startPosition);
+			} else if("dependency".equals(optionName)) {
```
Comment (lhein) [inline src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineOption.java:47]: I would also suggest to extract "dependency" into a constant.

File: src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java (hunk 1 above)
Comment (lhein) [reply src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java:65]: I think that the scheme must not fit the component name in any case.

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java:65]: Claus said that it is the component name https://chat.google.com/room/AAAAeFFdN0E/zrkZaoCUbAw
The component name is the scheme is in the catalog, isn't it?

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java:65]: hum maybe it is getLabel

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java:65]: or getLabel, getArtifactId, getScheme, getTitle... asked to Claus on the chat

Comment (lhein) [reply src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java:65]: image

Comment (lhein) [reply src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java:65]: was more thinking of the artifactId which would also remove the need to prepend the "camel-" string.

Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/modelinemodel/CamelKModelineDependencyOption.java:65]: so, Claus said it is the artifactId. let's go with that, well spotted

**Pull request merged**